### PR TITLE
code health rc5test

### PR DESCRIPTION
**Pull request by pauli-oracle**
- [x] tests are added or updated

Update rc5test to use the test infrastructure

### CODE REVIEW

File: test/rc5test.c [hunk 1]
```diff
@@ -187,90 +176,57 @@ static unsigned char rc5_cbc_iv[RC5_CBC_NUM][8] = {
...
-                printf("%02X ", buf[i]);
-            printf("\n");
-            printf("expected:");
-            for (i = 0; i < 8; i++)
-                printf("%02X ", RC5cipher[n][i]);
-            err = 20;
-            printf("\n");
-        }
+    RC5_32_ecb_encrypt(&(RC5plain[n][0]), buf, &key, RC5_ENCRYPT);
+    if (!TEST_mem_eq(&(RC5cipher[n][0]), 8, buf, 8))
```
Comment (mattcaswell) [inline test/rc5test.c]: sizeof(RC5cipher[0]) and sizeof(buf) instead of 8?

File: test/rc5test.c [hunk 2]
```diff
@@ -187,90 +176,57 @@ static unsigned char rc5_cbc_iv[RC5_CBC_NUM][8] = {
...
-            for (i = 0; i < 8; i++)
-                printf("%02X ", RC5plain[n][i]);
-            printf("\n");
-            err = 3;
-        }
-    }
-    if (err == 0)
-        printf("ecb RC5 ok\n");
+    RC5_32_ecb_encrypt(buf, buf2, &key, RC5_DECRYPT);
+    if (!TEST_mem_eq(&(RC5plain[n][0]), 8, buf2, 8))
```
Comment (mattcaswell) [inline test/rc5test.c]: Similarly here

File: test/rc5test.c [hunk 3]
```diff
@@ -187,90 +176,57 @@ static unsigned char rc5_cbc_iv[RC5_CBC_NUM][8] = {
...
-            for (i = 0; i < 8; i++)
-                printf("%02X ", buf[i]);
-            printf("\n");
-            printf("expected:");
-            for (i = 0; i < 8; i++)
-                printf("%02X ", rc5_cbc_cipher[n][i]);
-            err = 30;
-            printf("\n");
-        }
+        if (!TEST_mem_eq(&(rc5_cbc_cipher[n][0]), 8, buf, 8))
```
Comment (mattcaswell) [inline test/rc5test.c]: and here

**Comment by pauli-oracle** [comment]
sizeofs used instead of magic 8 constants.

**Comment by richsalz** [comment]
+2

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**